app/controllers/concerns/api/articles_controller.rb

```diff
@@ -123,6 +120,10 @@ def unpublish
 
     private
 
+    def per_page_max
+      (ApplicationConfig["API_PER_PAGE_MAX"] || 1000).to_i
+    end
```

Comment:
The reason I had to move this out of the file constant was because in the test suite once the file is evaluated (without the ENV variable stubbed) the constant is loaded with 1000. 

When we reach the test where we stub `ApplicationConfig["API_PER_PAGE_MAX"]` to a lower number it's never reevaluated and the tests fail.

I'd be willing to move it back up to a constant but I couldn't figure out a way to force reevaluate the classes, otherwise we need to do this in order to test this feature.